### PR TITLE
feat: add pantheon deployment workflow

### PR DESCRIPTION
**Pull request by theogainey**
Adds a workflow for deployment to Pantheon.

### Validate existing deployment

1. Verify successful workflow run triggered by this branch: https://github.com/sparkbox/sparkpress-wordpress-starter/actions/runs/6240893966
1. Visit the [pantheon dashboard](https://dashboard.pantheon.io/sites/d33d6067-e822-45a7-b431-87f1e4682fb1#dev/code)
1. View the commit log and verify that that there is a "Prod release" commit, created by this workflow.
1. Visit the [development site](http://dev-sparkpress.pantheonsite.io/) and verify that it loads with the SparkPress theme
1. Visit the [site admin](http://dev-sparkpress.pantheonsite.io/wp-admin/). 
1. Log into the admin using the `sparkpress` credentials in the Production Team vault in 1Password.
1. Verify that the admin works correctly.

### Validate documentation
There are a few quirks to setting up deployment for Pantheon. It's important to validate that the instructions for these are easy to follow. 

1. View the [Deployment docs](https://github.com/sparkbox/sparkpress-wordpress-starter/blob/d18ee8944b5fc6f7b78613750f871fd99f0b59eb/docs/deployment.md).
2. Verify that instructions are logically ordered and understandable
3. Validate steps 2-6 of the [site setup](https://github.com/sparkbox/sparkpress-wordpress-starter/blob/d18ee8944b5fc6f7b78613750f871fd99f0b59eb/docs/deployment.md#site-setup) instructions locally using the existing Pantheon site.
4. Validate instructions for [Obtaining PRODUCTION_REPO value](https://github.com/sparkbox/sparkpress-wordpress-starter/blob/d18ee8944b5fc6f7b78613750f871fd99f0b59eb/docs/deployment.md#obtaining-production_repo-value) to ensure that the instructions result in obtaining the correct value.
5. Validate instructions for [Obtaining KNOWN_HOSTS value](https://github.com/sparkbox/sparkpress-wordpress-starter/blob/d18ee8944b5fc6f7b78613750f871fd99f0b59eb/docs/deployment.md#obtaining-known_hosts-value) to ensure that the instructions result in obtaining the correct value.

### CODE REVIEW

**Comment by jonoliver** [comment]
Documenting this here for the moment:
In order for pantheon deployment to work, there needs to be a change made to the Pantheon remote's `wp-config.php` file. See this commit from Pantheon remote:
`67d2b5b37977ac5c7346d5b8f005e0236cd42bf9`

Change requires including the following line at the top of `wp-config.php`: 

```
require __DIR__ . '/vendor/autoload.php';
```

This is a difficult change to capture. In the future, we may want to maintain a pantheon version of `wp-config.php` that is copied over during the build.

File: .github/workflows/deploy.pantheon.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          git remote add pantheon ${{ secrets.PRODUCTION_REPO }}
+          git fetch pantheon master
+          git worktree add --track -b prod prod pantheon/master
+
+      - name: Copy build artifacts
+        run: |
+          rm -rf prod/vendor prod/wp-content/plugins prod/wp-content/themes/uga-ool
+          cp -r vendor prod
+          cp -r theme prod/wp-content/themes/uga-ool
+          cp -r wp-content/plugins prod/wp-content/plugins
```
Comment (dustin-jw) [inline .github/workflows/deploy.pantheon.yml:62]: We've got some copy-paste here with a few references to `uga-ool` that I think are meant to be `sparkpress`.

**Review by dustin-jw — CHANGES_REQUESTED** [review]
Everything's functioning as expected and the docs look good, I just noticed one spot where the naming is off.

File: .github/workflows/deploy.pantheon.yml (hunk 1 above)
Comment (jonoliver) [reply .github/workflows/deploy.pantheon.yml]: Whoops, thanks for catching that. Updated in 7c2391955dbfb125a7fb6f0d1cab309db606f03f

**Review by dustin-jw — APPROVED** [review]

**Comment by jonoliver** [comment]
Address #20

**Pull request merged**